Added heartbeats plug in and CustomCheck sample

seanfarmar opened the pull request.

mikeminutillo commented:
@seanfarmar what prompted this PR? Are you using this sample to demo stuff or was there an issue with the demo as is?

seanfarmar commented:
I'm using this as a demo for the platform and I was missing this...

seanfarmar commented:
I've updated this with v7 and updated ServiceControl and ServicePulse.

Happy to re PR this if you want separate PRs for each update?

mikeminutillo commented on Platform/servicecontrol/monitoring-instance/.diagnostics/Particular.Monitoring-configuration.txt:
```diff
@@ -0,0 +1 @@
+{"AssemblyScanning":{"Assemblies":["ServiceControl.Monitoring, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null","NServiceBus.NLog, Version=3.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c","NServiceBus.Core, Version=7.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c","NServiceBus.ObjectBuilder.Autofac, Version=7.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c","NServ ...
```
This folder should be ignored I think. We don't want to check in the diagnostics

seanfarmar replied:
pushed a fix

tmasternak commented:
@seanfarmar this is what I get running the pakackage:

``` PS 
Updating connection strings
Get-Content : Cannot find path 'C:\git\p\MonitoringDemo\Solution\binaries\CustomCheck\net461\CustomCheck.exe.config'
because it does not exist.
At C:\git\p\MonitoringDemo\localDB.ps1:69 char:16
+   $xml = [xml](Get-Content $ConfigFile)
+                ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\git\p\Monito...heck.exe.config:String) [Get-Content], ItemNotFoundEx
   ception
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetContentCommand

C:\git\p\MonitoringDemo\localDB.ps1 : Error starting setting up demo.
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,localDB.ps1

    + FullyQualifiedErrorId : {4}

 :
At C:\git\p\MonitoringDemo\localDB.ps1:70 char:3
+   $xml.SelectNodes("//connectionStrings/add") | % {
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
```

The build output for CustomCheck is missing from the repo - I think we need it.

Secondly I would propose to merge `3rdPartySystem` and `CustomCheck` into single project. Having two separate adds complexity to the project that users need to analyze and we only control the 3rdPartySystem http codes so single console for interaction should be enough.

seanfarmar commented:
@mikeminutillo @tmasternak I'll review this next week and fix, thanks

seanfarmar commented:
In the last push, I've moved custom checks to Shipping and removed the customCheck project (I'm hoping to add 3rd party integration on shipping to show saga monitoring). 
I also removed the .diagnostics folder form committed files

@tmasternak @mikeminutillo what do you think?

danielmarbach commented:
It is probably better to close this PR and restart with a clean slate given that so much has changed since
